### PR TITLE
Set QED support ON by default

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR is very similar  to https://github.com/ECP-WarpX/WarpX/pull/1486 (but less radical, since it is still possible to compile without QED support)

**warning:** do not merge! Merging must be done simultaneously with https://github.com/ECP-WarpX/picsar/pull/7

### CODE REVIEW

**Comment by ax3l** [comment]
Scratching my head what's off here... We might need to schedule a VC on it to debug it together... Will try locally for now.

Update: the extra includes and missing prefixes for picsar are causing this, mainly the collision with cuda math include names (`math_constants.h`). We will prefix everything with `picsar/` and only add one `target_include_directory` to its root location to make this more robust on the long-term.

**Comment by ax3l** [comment]
Hi @lucafedeli88, let's find a time slot for a VC in the mornings to progress on this PR :)

File: cmake/dependencies/PICSAR.cmake [hunk 1]
```diff
@@ -41,10 +61,10 @@ endfunction()
 
 if(WarpX_QED)
     option(WarpX_picsar_internal   "Download & build PICSAR" ON)
-    set(WarpX_picsar_repo "https://github.com/ECP-WarpX/picsar.git"
+    set(WarpX_picsar_repo "https://github.com/lucafedeli88/picsar.git"
         CACHE STRING
         "Repository URI to pull and build PICSAR from if(WarpX_picsar_internal)")
-    set(WarpX_picsar_branch "development"
+    set(WarpX_picsar_branch "improve_makefile"
```
Comment (ax3l) [inline cmake/dependencies/PICSAR.cmake]: Memo to ourselves: let's not forget to wait for the upstream PR to be merged and then revert this change here

**Comment by NeilZaim** [comment]
I have no specific comment on this PR, LGTM. :)  As in the picsar one, I'd let @ax3l decide if this is ready for merging.

(without forgetting to revert to picsar development branch in `PICSAR.cmake` and `run_tests.sh`).

**Comment by ax3l** [comment]
triggering CI due to a change in it

**Comment by lucafedeli88** [comment]
@ax3l , do you think that this PR is ready to be merged?

File: run_test.sh [hunk 2]
```diff
@@ -50,11 +50,7 @@ echo "cd $PWD"
 # Clone PICSAR and AMReX
 git clone --branch development https://github.com/AMReX-Codes/amrex.git
 # Use QED brach for QED tests
-if [ "${WARPX_CI_QED}" = "TRUE" ]; then
-    git clone --branch development https://github.com/ECP-WarpX/picsar.git
-else
-    git clone --branch development https://github.com/ECP-WarpX/picsar.git
-fi
+git clone --branch improve_makefile https://github.com/lucafedeli88/picsar.git
```
Comment (ax3l) [inline run_test.sh]: Memo to ourselves: let's not forget to wait for the upstream PR to be merged and then revert this change here
```suggestion
git clone --branch development https://github.com/ECP-WarpX/picsar.git
```

**Review by ax3l — APPROVED** [review]
LGTM - do you like to coordinate the merge in PICSAR and here?

**Comment by lucafedeli88** [comment]
> LGTM - do you like to coordinate the merge in PICSAR and here?

Sure! I can do that... however, I think that you have to approve https://github.com/ECP-WarpX/picsar/pull/7 before I can merge it. As soon as I can do that I will merge the PR in PICSAR and I will update and merge this one.

File: cmake/dependencies/PICSAR.cmake [hunk 3]
```diff
@@ -41,10 +65,10 @@ endfunction()
 
 if(WarpX_QED)
     option(WarpX_picsar_internal   "Download & build PICSAR" ON)
-    set(WarpX_picsar_repo "https://github.com/ECP-WarpX/picsar.git"
+    set(WarpX_picsar_repo "https://github.com/lucafedeli88/picsar.git"
```
Comment (ax3l) [inline cmake/dependencies/PICSAR.cmake]:
```suggestion
    set(WarpX_picsar_repo "https://github.com/ECP-WarpX/picsar.git"
```

File: cmake/dependencies/PICSAR.cmake [hunk 4]
```diff
@@ -41,10 +65,10 @@ endfunction()
...
         CACHE STRING
         "Repository URI to pull and build PICSAR from if(WarpX_picsar_internal)")
-    set(WarpX_picsar_branch "development"
+    set(WarpX_picsar_branch "improve_makefile"
```
Comment (ax3l) [inline cmake/dependencies/PICSAR.cmake]:
```suggestion
    set(WarpX_picsar_branch "development"
```

**Pull request merged**